docs/release_checklist.md

```diff
@@ -0,0 +1,12 @@
+# Release checklist
+
+NOTE: Fix problems by moving forward (roll a new release with a fix).
+
+- [ ] Create release branch (`*.*.*`)
```

Comment:
- [ ] Create new release branch, branched from develop, using the convention `feature/vx.x.x`, were `x.x.x` is the new version based on [Semver](https://semver.org/).

Explain where and how the package gets published.
Explain how to check published packages and deployments (link to packages to make it easier)